homeassistant/components/webostv/__init__.py

```diff
@@ -83,8 +88,44 @@
 
 
 async def async_setup(hass, config):
-    """Set up the LG WebOS TV platform."""
-    hass.data[DOMAIN] = {}
+    """Set up the environment."""
+    hass.data.setdefault(DOMAIN, {})
+    if DOMAIN not in config:
+        return True
+
+    for conf in config[DOMAIN]:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set the config entry up."""
+    if not config_entry.options:
+        config = config_entry.data
+        options = {}
+
+        # Get Turn_on service
+        options[CONF_ON_ACTION] = config.get(CONF_ON_ACTION)
+
+        # Get Preferred Sources
+        sources = config.get(CONF_CUSTOMIZE, {}).get(CONF_SOURCES)
+        if sources:
```

Comment:
You could use a walrus operator here

 'if sources := ' since we require python 3.8+